bundles/net.tourbook/src/net/tourbook/messages.properties

```diff
@@ -2333,9 +2333,13 @@ Pref_Weather_Check_SaveLogInTourWeatherDescription_Tooltip = When checked, all t
 Pref_Weather_Checkbox_ShowOrHideApiKey                     = &Show/hide API key
 Pref_Weather_Checkbox_ShowOrHideApiKey_Tooltip             = Toggles between hidden or fully displayed API Key
 Pref_Weather_Label_ApiKey                                  = &API Key
-Pref_Weather_Label_OpenWeatherMap_FiveDaysLimit            = NOTE:\n\
-                                                             While the weather retrieval with this vendor is free,\n\
-                                                             historical weather data can only be retrieved for the previous 5 days.
+Pref_Weather_Label_OpenWeatherMap_TimeRangeLimit           = NOTE:\n\
+                                                             While the weather retrieval for this vendor is free and \n\
+                                                             the historical weather data can be retrieved for the last \n\
+                                                             40+ years (from 1st January 1979), the total number of \n\
+                                                             requests per day is limited to 1000. Please be mindful of \n\
+                                                             this as this total number of daily requests is shared across \n\
+                                                             all the users.
```

Comment:
The text in text/label controls should always grab the whole available horizontal space...

![image](https://github.com/mytourbook/mytourbook/assets/1283445/4468fdf9-f57b-45de-9e2a-35733423973e)

...like in the import page...

![image](https://github.com/mytourbook/mytourbook/assets/1283445/a201c5f0-7caa-49f7-9f11-accdc6a59ba0)

...only the tooltip text should be wrapped manually

Reply:
Fixed

![image](https://github.com/mytourbook/mytourbook/assets/25235665/82aaf701-b150-47ce-9c74-e3877459c599)

Reply:
After thinking more about the request max limit, we should limit the request per day/session because with just a few clicks in MT, you can easily intentionally or accidentally reach the max. limit.

![image](https://github.com/mytourbook/mytourbook/assets/1283445/2ca77ed4-c73d-4160-b8e8-18ccb1fd19f4)

The wording "Retrieve Weather Data..." is irritating because "..." means that there will be a dialogbox to setup/confirm this action but in this case, the weather retrieval starts immediately

Reply:
> After thinking more about the request max limit, we should limit the request per day/session because with just a few clicks in MT, you can easily intentionally or accidentally reach the max. limit.

That is an idea yes. What do you think the limit should be per user per day ?

What mechanism do you have in mind in order to save the number of requests made per day, using the state like [this one](https://github.com/FJBDev/mytourbook/blob/c14dfd2c98a4f8033e40958d3c2545bdcae55e5b/bundles/net.tourbook/src/net/tourbook/ui/views/nutrition/TourNutritionView.java#L143)?
> 

> 
> The wording "Retrieve Weather Data..." is irritating because "..." means that there will be a dialogbox to setup/confirm this action but in this case, the weather retrieval starts immediately

It takes more to irritate me 😄  but I can change it if you want, to `Retrieve Weather Data` without the `...`

Reply:
> > After thinking more about the request max limit, we should limit the request per day/session because with just a few clicks in MT, you can easily intentionally or accidentally reach the max. limit.
> 
> That is an idea yes. What do you think the limit should be per user per day ?

20 ... 40 ? 

I think this would be enough when a few tour days are not imported and is mostly more than the current limit of 5 days

Reply:
> > After thinking more about the request max limit, we should limit the request per day/session because with just a few clicks in MT, you can easily intentionally or accidentally reach the max. limit.
> 
> What mechanism do you have in mind in order to save the number of requests made per day, using the state like [this one](https://github.com/FJBDev/mytourbook/blob/c14dfd2c98a4f8033e40958d3c2545bdcae55e5b/bundles/net.tourbook/src/net/tourbook/ui/views/nutrition/TourNutritionView.java#L143)?

When not necessary, I prefer the state. In the beginning of MT, I only used the pref store because I was not aware of the states, both have advantages/disadvantages

Reply:
> After thinking more about the request max limit, we should limit the request per day/session because with just a few clicks in MT, you can easily intentionally or accidentally reach the max. limit.
> 

Suggestion implemented. I have decided to limit each user to 50 requests per day. I chose 50 requests because for a given tour, a request will be made for each hour of the tour and therefore retrieving data for several tours can quickly request multiple requests

![image](https://github.com/mytourbook/mytourbook/assets/25235665/277221c5-2459-4377-adc9-1e3e5e7a2305)


Preference message updated to show to the user the 50 max requests per day
![image](https://github.com/mytourbook/mytourbook/assets/25235665/1186a7ef-87bb-4bd2-ac05-4e08665e1e62)